src/state-of-the-layer/state-of-the-layer.js

```diff
@@ -0,0 +1,290 @@
+                    var theKey = decodeURIComponent(subItems[0]),
+                        theValue = decodeURIComponent(subItems[1]);
+
+                    // If we haven't had this key already, add it to the current state
+                    if (this.keyOrder.indexOf(theKey) == -1) {
+                        this.state[theKey] = theValue;
+                        this.keyOrder.push(theKey);
+                    }
+
+                }
+            }
+        }
+    };
+
+    UrlHashManager.prototype = {
+
+        /**
+         * Get Studio tags required to activated to represent current state
+         *
+         * @returns {Array}
+         */
+        getTagsForCurrentState: function(){
+
+            var tags = [];
+
+            // Build array of tags
+            for (var i = 0; i < this.keyOrder.length; i++) {
+                var theKey = this.keyOrder[i];
+                tags.push(theKey + "=" + this.state[theKey]);
+            }
+
+            return tags;
+        },
+
+        /**
+         * Build a string representing the current state for use in the URL
+         *
+         * @returns {string}
+         */
+        getUrlHash: function() {
+
+            var value = "";
+
+            for (var i = 0; i < this.keyOrder.length; i++) {
+
+                var theKey = this.keyOrder[i];
+
+                if (i != 0) {
+                    value += "&";
+                }
+
+                // Encode values to make the URL safe
+                value += encodeURIComponent(theKey) + "=" + encodeURIComponent(this.state[theKey]);
+            }
+
+            return value;
+        },
+
+        /**
+         * Set item for use in URL
+         *
+         * @param {string} theKey
+         * @param {string} theValue
+         */
+        setItemValue: function(theKey, theValue) {
+
+            // If we don't have this item already
+            if (this.keyOrder.indexOf(theKey) == -1) {
+                this.keyOrder.push(theKey);
+            }
+
+            this.state[theKey] = theValue;
+
+            this.updateUrl();
+        },
+
+        /**
+         * Apply the current hash to the URL
+         */
+        updateUrl: function() {
+            window.location.hash = this.getUrlHash();
+        }
+    };
+
+    /**
+     * Global object to manage the Experience's state represented in the URL
+     *
+     * @type {UrlHashManager}
+     */
+    var urlManager = new UrlHashManager();
+
+    /**
+     * The Experience's ID, or NULL if Standalone
+     *
+     * @type {string|null}
+     */
+    var experienceId = null;
+
+    /**
+     * Function that runs when both DOM and SDK are ready
+     */
+    var activatePlugin = function() {
+
+        // Find the request, or current, Experience
+        CerosSDK.findExperience(experienceId).done(function(experience) {
+
+            // Find items that set the state
+            var componentCollection = experience.findComponentsByTag("set-state");
```

Comment:
this whole chunk could be in a named private function explaining that this will add relevant payloads to hash